indexing-hadoop/src/main/java/io/druid/indexer/IndexGeneratorJob.java

```diff
@@ -520,7 +530,7 @@ protected void reduce(
         int runningTotalLineCount = 0;
         long startTime = System.currentTimeMillis();
 
-        Set<String> allDimensionNames = Sets.newHashSet();
+        Set<String> allDimensionNames = Sets.newLinkedHashSet();
```

Comment:
@jon-wei does it always return the same dimensions for every single row, and is that also the case for other InputRowParsers.